tasks/test_results_processor.py

```diff
@@ -107,167 +126,147 @@ def _bulk_write_tests_to_db(
-                testsuite = testrun.testsuite
-                outcome = str(testrun.outcome)
-                duration_seconds = testrun.duration
-                failure_message = testrun.failure_message
-                test_id = generate_test_id(repoid, testsuite, name, flags_hash)
-
-                test_data.append(
-                    dict(
-                        id=test_id,
-                        repoid=repoid,
-                        name=name,
-                        testsuite=testsuite,
-                        flags_hash=flags_hash,
-                    )
+        test_data = []
+        test_instance_data = []
+        daily_totals = dict()
+        for testrun in parsed_testruns:
+            # Build up the data for bulk insert
+            name = testrun.name
+            testsuite = testrun.testsuite
+            outcome = str(testrun.outcome)
+            duration_seconds = testrun.duration
+            failure_message = testrun.failure_message
+            test_id = generate_test_id(repoid, testsuite, name, flags_hash)
+
+            test_data.append(
+                dict(
+                    id=test_id,
+                    repoid=repoid,
+                    name=name,
+                    testsuite=testsuite,
+                    flags_hash=flags_hash,
                 )
+            )
 
-                test_instance_data.append(
-                    dict(
-                        test_id=test_id,
-                        upload_id=upload_id,
-                        duration_seconds=duration_seconds,
-                        outcome=outcome,
-                        failure_message=failure_message,
-                        commitid=commitid,
-                        branch=branch,
-                        reduced_error_id=None,
-                        repoid=repoid,
-                    )
+            test_instance_data.append(
+                dict(
+                    test_id=test_id,
+                    upload_id=upload_id,
+                    duration_seconds=duration_seconds,
+                    outcome=outcome,
+                    failure_message=failure_message,
+                    commitid=commitid,
+                    branch=branch,
+                    reduced_error_id=None,
+                    repoid=repoid,
                 )
+            )
 
-                def update_daily_total():
-                    daily_totals[test_id]["last_duration_seconds"] = duration_seconds
-                    daily_totals[test_id]["avg_duration_seconds"] = (
-                        daily_totals[test_id]["avg_duration_seconds"]
-                        * (
-                            daily_totals[test_id]["pass_count"]
-                            + daily_totals[test_id]["fail_count"]
-                        )
-                        + duration_seconds
-                    ) / (
+            def update_daily_total():
+                daily_totals[test_id]["last_duration_seconds"] = duration_seconds
```

Comment:
Can we add a comment here for what the formula is? It's a bit hard to parse otherwise